webapp/resources/src/main/webapp/javascript/eXo/UISpaceInfoPopup.js

```diff
@@ -234,45 +234,43 @@
-                                            }
-                                            if(membersData){
-                                              spaceData.member = membersData.size;
-                                            } else {
-                                              spaceData.member =0;
-                                            }
-                                            if(managerData){
-                                              spaceData.manager = managerData.size;
-                                            }else {
-                                              spaceData.manager =0;
-                                            }
-
-                                            spaceData.membership = rolesArray;
-
-                                            buildPopup(spaceData, spaceUrl);
-                                            putToCache(spaceId, spaceData);
+                                        var rolesArray = {
+                                          roles: []
+                                        };
+                                        var currentUser = membersData.users.find(user => {
+                                          return user.username === eXo.env.portal.userName
+                                        })
+                                        if(currentUser){
+                                          if (currentUser.isManager){
+                                            rolesArray.roles.push("manager");
                                           }
-                                        });
+                                          if (currentUser.isMember){
+                                            rolesArray.roles.push("member");
+                                          }
+                                        }
+                                        if(!spaceData.avatarUrl){
+                                          spaceData.avatarUrl= opts.defaultAvatarUrl;
+                                        }
+                                        if(membersData){
+                                          spaceData.member = membersData.size;
+                                        } else {
+                                          spaceData.member =0;
+                                        }
+                                        if(membersData){
+                                          spaceData.member = membersData.size;
+                                        } else {
+                                          spaceData.member =0;
+                                        }
+                                        if(managerData){
+                                          spaceData.manager = managerData.size;
+                                        }else {
+                                          spaceData.manager =0;
+                                        }
```

Comment:
Could you refactor this code and optimize it. I know it is old, let's improve it ;-)

Reply:
done!